test/integration_test.go

```diff
@@ -6174,6 +6174,91 @@ func (ts *IntegrationTestSuite) TestScheduleUpdateWorkflowActionMemo() {
 	}
 }
 
+func (ts *IntegrationTestSuite) TestVersioningBehaviorInRespondWorkflowTaskCompletedRequest() {
+	versioningBehaviorAll := make([]workflow.VersioningBehavior, 0)
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+
+	// We are setting the default build ID with versioning-2 rules to test
+	// with existing servers. TODO(antlai-temporal) use versioning-3 APIs
+	// after there is a server release that supports versioning-3
+	res, err := ts.client.GetWorkerVersioningRules(ctx, client.GetWorkerVersioningOptions{
+		TaskQueue: ts.taskQueueName,
+	})
+	ts.NoError(err)
+
+	_, err = ts.client.UpdateWorkerVersioningRules(ctx, client.UpdateWorkerVersioningRulesOptions{
+		TaskQueue:     ts.taskQueueName,
+		ConflictToken: res.ConflictToken,
+		Operation: &client.VersioningOperationInsertAssignmentRule{
+			RuleIndex: 0,
+			Rule: client.VersioningAssignmentRule{
+				TargetBuildID: "1.0",
+			},
+		},
+	})
+	ts.NoError(err)
+
+	c, err := client.Dial(client.Options{
+		HostPort:  ts.config.ServiceAddr,
+		Namespace: ts.config.Namespace,
+		ConnectionOptions: client.ConnectionOptions{
+			TLS: ts.config.TLS,
+			DialOptions: []grpc.DialOption{
+				grpc.WithUnaryInterceptor(func(
+					ctx context.Context,
+					method string,
+					req interface{},
+					reply interface{},
+					cc *grpc.ClientConn,
+					invoker grpc.UnaryInvoker,
+					opts ...grpc.CallOption,
+				) error {
+					if method == "/temporal.api.workflowservice.v1.WorkflowService/RespondWorkflowTaskCompleted" {
+						asReq := req.(*workflowservice.RespondWorkflowTaskCompletedRequest)
+						behavior := internal.VersioningBehaviorFromProto(asReq.VersioningBehavior)
```

Comment:
This seems to be the only use of the "from proto" exposed utility, but there doesn't seem to be much benefit converting in this test. Can you just use the raw proto enum in this test for this slice and not have the utility?

Reply:
I think the bigger question is your previous comment, whether we just expose the enum protos and be done with it. If not I think sooner or later users may want to go back and forth, so more boiler plate...

Reply:
Ok, I don't think implementing the FromProto if they need it is a big deal. I'm getting rid of it, using proto in the test, and making the toProto private.

Reply:
Works for me. Yeah the high-level wrapper is nicer for users, and no need to convert the other way.